packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -445,6 +509,25 @@ const SideNav = ({
     [sideNavExpandClickHandler],
   );
 
+  const setSortedItems = (
+    parentId: string,
+    activeIndex: number,
+    overIndex: number,
+  ) => {
+    const newSideNavItems = sideNavItemsList.map((item) => {
+      if (item.id === parentId && item.nestedNavItems) {
+        item.nestedNavItems = arrayMove(
+          item.nestedNavItems,
+          activeIndex,
+          overIndex,
+        );
+      }
+      return item;
+    });
+    updateSideNavItemsList(newSideNavItems);
+    onSort?.(newSideNavItems);
+  };
```

Comment:
We shouldn't be mutating values we get in. That's going to cause issues and is a general bad practice.

Also, this function needs to be wrapped in `useCallback`.

```suggestion
  const setSortedItems = (
    parentId: string,
    activeIndex: number,
    overIndex: number,
  ) => {
    const newSideNavItems = sideNavItemsList.map((item) => {
      if (item.id === parentId && item.nestedNavItems) {
        return {
          ...item,
          item.nestedNavItems: arrayMove(
            item.nestedNavItems,
            activeIndex,
            overIndex,
          )
        }
      }
      
      return item;
    });
    updateSideNavItemsList(newSideNavItems);
    onSort?.(newSideNavItems);
  };
```

Reply:
We shouldn't be mutating the values of `item`. `sideNavItemsList` should be pure, and by mutating values given to us in `.map`, we're not abiding by the expected contract of `.map`.

```suggestion
const setSortedItems = useCallback((
  parentId: string,
  activeIndex: number,
  overIndex: number,
) => {
  const sortedSideNavItems = sideNavItemsList.map((item) => ({
    ...item,
    nestedNavItems: (
      item.id === parentId && item.nestedNavItems
      ? (
        arrayMove(
          item.nestedNavItems,
          activeIndex,
          overIndex,
        )
      )
      : undefined
    )
  });
  
  updateSideNavItemsList(sortedSideNavItems);
  onSort?.(sortedSideNavItems);
}, [onSort, sideNavItemsList, updateSideNavItemsList]);
```

Reply:
If this doesn't work because of the `undefined`:
```ts
    nestedNavItems: (
      item.id === parentId && item.nestedNavItems
      ? (
        arrayMove(
          item.nestedNavItems,
          activeIndex,
          overIndex,
        )
      )
      : undefined
    )
```

Then it needs to be this:
```ts
...(
  item.id === parentId && item.nestedNavItems
  ? {
    nestedNavItems: (
      arrayMove(
        item.nestedNavItems,
        activeIndex,
        overIndex,
      )
    )
  }
  : {}
)
```

The latter will play nicer with TypeScript even if it reads messier.